jax/core.py

```diff
@@ -2050,104 +2053,139 @@ def check_map(prim, in_avals, params):
 
 
 # ------------------- Jaxpr printed representation -------------------
-def pp_vars(vs: Sequence[Any], *, print_shapes: bool = False) -> pp.Doc:
+
+# A JaxprPpContext allow us to uniquify variable names within nested Jaxprs.
+class JaxprPpContext:
```

Comment:
We could consider replacing `JaxprPpContext` with a `defaultdict(itertools.count().__next__)`. That would mean we don't need the logic about incrementing `num_vars` etc, and we could just use indexing syntax in `pp_var`.

Reply:
Done, that's a neat trick I didn't know.